lightning/src/ln/channelmanager.rs

```diff
@@ -2731,6 +2738,43 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// Send a payment that is probing the given route for liquidity. We calculate the
+	/// [`PaymentHash`] of probes based on a static secret and a random [`PaymentId`], which allows
+	/// us to easily discern them from real payments. This can be checked by calling
+	/// [`payment_is_probe`].
+	///
+	/// [`payment_is_probe`]: Self::payment_is_probe
+	pub fn send_probe_payment(&self, route: &Route) -> Result<(PaymentHash, PaymentId), PaymentSendFailure> {
+		let payment_id = PaymentId(self.keys_manager.get_secure_random_bytes());
+
+		let mut preimage = [0u8; 64];
+		preimage[..32].copy_from_slice(&self.probing_cookie_secret);
+		preimage[32..].copy_from_slice(&payment_id.0);
+
+		let payment_hash = PaymentHash(Sha256::hash(&preimage).into_inner());
```

Comment:
Consider adding a helper for this since it is repeated later. Or rename `payment_is_probe` and have it return the payment hash instead of comparing.

Reply:
Done with https://github.com/lightningdevkit/rust-lightning/pull/1567/commits/b4cc28533ebcac2fce97b18b2d10c4148c7aa868.